libcloud/compute/drivers/kubevirt.py

```diff
@@ -0,0 +1,996 @@
+        :type accessMode: `str`
+
+        :param matchLabels: A dictionary with the labels, ie:
+                            {'release': 'stable,}
+        :type matchLabels: `dict` with keys `str` and values `str`
+        """
+        pvc = {
+            'apiVersion': 'v1',
+            'kind': 'PersistentVolumeClaim',
+            'metadata': {
+                'name': name
+            },
+            'spec': {
+                'accessModes': [],
+                'volumeMode': volume_mode,
+                'resources': {
+                    'requests': {
+                        'storage': ''
+                    }
+                },
+                'selector': {}
+            }
+        }
+
+        pvc['spec']['accessModes'].append(access_mode)
+
+        if storage_class_name is not None:
+            pvc['spec']['storageClassName'] = storage_class_name
+        else:
+            raise ValueError("The storage class name must be provided of a"
+                             "storage class which allows for dynamic "
+                             "provisioning")
+        pvc['spec']['resources']['requests']['storage'] = str(size) + 'Gi'
+
+        method = "POST"
+        req = ROOT_URL + "namespaces/" + namespace + "/persistentvolumeclaims"
+        data = json.dumps(pvc)
+        try:
+            result = self.connection.request(req, method=method, data=data)
+        except Exception as exc:
+            raise
+        if result.object['status']['phase'] != "Bound":
+            for _ in range(3):
+
+                req = ROOT_URL + "namespaces/" + namespace + \
+                    "/persistentvolumeclaims/" + name
+                try:
+                    result = self.connection.request(req).object
+                except Exception as exc:
+                    raise
+                if result['status']['phase'] == "Bound":
+                    break
+                time.sleep(3)
+
+        # check that the pv was created and bound
+        volumes = self.list_volumes()
+        for volume in volumes:
+            if volume.extra['pvc']['name'] == name:
+                return volume
+
+    def destroy_volume(self, volume):
+        # first delete the pvc
+        if volume.extra['isBound']:
+            pvc = volume.extra['pvc']['name']
+            namespace = volume.extra['pvc']['namespace']
+            method = 'DELETE'
+            req = ROOT_URL + "namespaces/" + namespace + \
+                "/persistentvolumeclaims/" + pvc
+            try:
+                result = self.connection.request(req, method=method)
+
+            except Exception as exc:
+                raise
+
+        pv = volume.name
+        req = ROOT_URL + "persistentvolumes/" + pv
+
+        try:
+            result = self.connection.request(req, method=method)
+            return result.status
+        except Exception as exc:
+            raise
+
+    def attach_volume(self, volume, ex_node, **kwargs):
```

Comment:
Please make sure the method signature matches the base API one and that the non standard arguments are prefixed with ``ex_``.

Base signature looks like this:

```python
def attach_volume(self, node, volume, device=None):
```

So ``node`` argument needs to be first, etc.